saleor/static/dashboard-next/categories/views/CategoryDetails.tsx

```diff
@@ -65,101 +70,94 @@ interface CategoryPaginationProviderProps {
+  const PAGINATE_BY = 20;
+  const paginationState = createPaginationState(PAGINATE_BY, params);
+  return (
+    <TypedCategoryPropertiesQuery
+      query={categoryPropertiesQuery}
+      variables={{ id, ...paginationState }}
+      fetchPolicy="network-only"
+    >
+      {({ loading, error, data }) => {
+        if (error) {
+          return (
+            <ErrorMessageCard
+              message={i18n.t("Unable to find a matching category.")}
+            />
+          );
+        }
+        const {
+          loadNextPage,
+          loadPreviousPage,
+          pageInfo
+        } = createPaginationData(
+          navigate,
+          paginationState,
+          categoryShowUrl(id),
+          data && data.category && data.category.products
+            ? data.category.products.pageInfo
+            : undefined,
+          loading
         );
-      }
-      const { category } = data;
-      const updatePageInfo = (
-        results: CategoryPropertiesQuery,
-        overrides: Partial<
-          CategoryPropertiesQuery["category"]["products"]["pageInfo"]
-        >
-      ) => ({
-        ...results,
-        category: {
-          ...results.category,
-          products: {
-            ...results.category.products,
-            pageInfo: {
-              ...results.category.products.pageInfo,
-              ...overrides
-            }
-          }
+
+        if (typeof children === "function") {
+          return children({
+            data,
+            loadNextPage,
+            loadPreviousPage,
+            loading,
+            pageInfo
+          });
         }
-      });
-      const fetchNextPage = () => {
-        return fetchMore({
-          updateQuery: (previousResult, { fetchMoreResult }) =>
-            updatePageInfo(fetchMoreResult, {
-              hasPreviousPage: true
-            }),
-          variables: {
-            after: category.products.pageInfo.endCursor,
-            first: 12
-          }
-        });
-      };
-      const fetchPreviousPage = () => {
-        return fetchMore({
-          updateQuery: (previousResult, { fetchMoreResult }) =>
-            updatePageInfo(fetchMoreResult, {
-              hasNextPage: true
-            }),
-          variables: {
-            before: category.products.pageInfo.startCursor,
-            first: undefined,
-            last: 12
-          }
-        });
-      };
-      if (typeof children === "function") {
-        return children({ data, loading, fetchNextPage, fetchPreviousPage });
-      }
-      if (React.Children.count(children) > 0) {
-        return React.Children.only(children);
-      }
-      return null;
-    }}
-  </TypedCategoryPropertiesQuery>
-);
+        if (React.Children.count(children) > 0) {
+          return React.Children.only(children);
+        }
+        return null;
+      }}
+    </TypedCategoryPropertiesQuery>
+  );
+};
 
 interface CategoryDetailsProps {
-  filters: any;
+  params: any;
```

Comment:
Please validate input and select appropriate keys in router where `qs` is parsed.